starlette/requests.py

```diff
@@ -199,8 +199,8 @@ def receive(self) -> Receive:
         return self._receive
 
     async def stream(self) -> typing.AsyncGenerator[bytes, None]:
-        if hasattr(self, "_body"):
-            yield self._body
+        if "body" in self.scope:
+            yield self.scope["body"]
```

Comment:
HTTPScope doesn't have a `["body"]` key: https://github.com/django/asgiref/blob/main/asgiref/typing.py#L65-L78

Reply:
i'm guessing the correct thing to do here then is to set the body inside of `scope['extensions']`? i'm not sure what the namespace for it should be. request_cache? `self.scope['extensions']['request_cache']['body']`?

https://asgi.readthedocs.io/en/latest/specs/main.html#extensions

Reply:
moved it to `scope["extensions"]["starlette"]["stream"]`